course05/homework02/freekpaans/linked_list.cc

```diff
@@ -0,0 +1,74 @@
+#include <string>
+#include <numeric>
+
+#include "linked_list.h"
+
+static void push_at(LinkedList::LinkPtr& link, const std::string& str)
+{
+	LinkedList::LinkPtr newLink{new LinkedList::Link { str, {} }};
+
+	newLink->mNext = std::move(link);
+	link = std::move(newLink);
+}
```

Comment:
This looks like Link's missing constructor code :)

Reply:
Not really following, are you suggesting doing the last assignment in the ctor as well?